src/D2L.CodeStyle.Analyzers/ApiUsage/RpcAnalyzer.cs

```diff
@@ -102,6 +112,27 @@ INamedTypeSymbol dependencyAttributeType
 				// other things to check:
 				// - appropriate use of [Dependency]
 			}
+
+			CheckThatRpcParametersAreValid( context, method, deserializerType );
+		}
+
+		private static void CheckThatRpcParametersAreValid(
+			SyntaxNodeAnalysisContext context,
+			MethodDeclarationSyntax method,
+			INamedTypeSymbol deserializerType
+		) {
+			if( deserializerType.GetType() == null ) {
+				throw new Exception( "Failed to resolve type: 'D2L.Serialization.IDeserializer'" );
+			}
+			var parameters = method.ParameterList.Parameters;
+			foreach( var parameterType in parameters.Select( p => p.Type ).Where( parameterType => parameterType != null ) ) {
+				bool isValidParameterType = WebpagesRpcParameterTypeValidator.IsValidParameterType( type: parameterType.GetType(), deserializerType: deserializerType.GetType() );

```

Comment:
`GetType()` won't work here (see other thing about reflection types vs. semantic model types.)
If you want to chat about Roslyn APIs feel free to DM me on Slack.